lapack/testlapack/general.go

```diff
@@ -828,18 +828,16 @@ func isOrthogonal(q blas64.General) bool {
 	// an orthonormal basis of the Euclidean space R^n.
 	const tol = 1e-13
 	for i := 0; i < n; i++ {
-		nrm := blas64.Nrm2(n, blas64.Vector{Data: q.Data[i*q.Stride:], Inc: 1})
+		nrm := blas64.Nrm2(blas64.Vector{N: n, Data: q.Data[i*q.Stride:], Inc: 1})
 		if math.IsNaN(nrm) {
 			return false
 		}
 		if math.Abs(nrm-1) > tol {
 			return false
 		}
 		for j := i + 1; j < n; j++ {
-			dot := blas64.Dot(n,
-				blas64.Vector{Data: q.Data[i*q.Stride:], Inc: 1},
-				blas64.Vector{Data: q.Data[j*q.Stride:], Inc: 1},
-			)
+			dot := blas64.Dot(blas64.Vector{Data: q.Data[i*q.Stride:], Inc: 1},
```

Comment:
Where is `n`?

Reply:
This is the foot gun that exists with this approach.

Reply:
Agreed. The previous foot gun was having one size though the two vectors were the same size. Unfortunately, this is a problem with testing Lapack, which doesn't get the niceties `mat` does. There are many fewer hand-coded `blas.Vector` in `mat`